src/parenthetics.py

```diff
@@ -0,0 +1,21 @@
+"""Determines whether a string has balanced, open, or closed parens."""
+from stack import Stack
+
+
+def eval_parens(input):
+    """Evaluate whether a string has balanced, open, or closed parens."""
+    print('input type:', type(input))
+    if type(input) != str:
+        raise TypeError('Please enter a string.')
+    if '(' not in input and ')' not in input:
```

Comment:
Updated - I was having a problem earlier with that statement but now it passes all the tests.